### PR TITLE
Add Apple Silicon support & major revamp (3.0.0)

### PR DESCRIPTION
**Pull request by waydabber**
This request is about moving 3.0.0 changes to master as discussed.

### CODE REVIEW

File: MonitorControl/Model/ExternalDisplay.swift
Comment (JoniVR) [inline MonitorControl/Model/ExternalDisplay.swift:57]:
```suggestion
    if !isVirtual && !Arm64DDCUtils.isArm64 {
      self.ddc = DDC(for: identifier)
```

File: MonitorControl/View Controllers/AboutPrefsViewController.swift
Comment (JoniVR) [inline MonitorControl/View Controllers/AboutPrefsViewController.swift:56]:
```suggestion
    self.versionLabel.stringValue = "\(versionName) \(versionNumber) \(buildName) \(buildNumber) - \(arch)"
```

Should we mark RC as well?

File: README.md
Comment (JoniVR) [inline README.md:51]: Perhaps a note for users with older versions on which version they can use?

**Review by JoniVR — COMMENTED** [review]
I've done a quick run through the PR to get in some early feedback.

Most of the things I've noted are just questions or some really minor stuff, I've probably missed some stuff due to the huge amount of changes, but at first sight everything looks good to me.

The amount of code you've produced in this short of a timespan is simply astonishing (🤯), and I think the end result is pretty amazing. You should be really proud of this. Some really advanced and impressive implementations in here.

As a side note, some of the images inside .github are not being used anymore either, we can either keep them or get rid of them, your call :)

I'll try to get around to the Intel side this weekend if I get the time, together with a first release candidate, if I can't get Intel working properly or don't have the time for it I'll just go with the optimisation setting to buy some extra time on the fix.

File: MonitorControl/AppDelegate.swift
Comment (waydabber) [reply MonitorControl/AppDelegate.swift:23]: It is just a debug switch but very useful for me so if it doesn't bother anybody I'd like to keep it for now. This quickly makes InternalDisplay and ExternalDisplay for me and converts every display to force-software controlled. After a while we might get rid of it when the Display refactoring is done.

File: MonitorControl/AppDelegate.swift
Comment (waydabber) [reply MonitorControl/AppDelegate.swift:112]: This should be a valid way to vet DDC capable displays but some DDC capable displays fail on this. I wanted to improve on that a bit so under Displays there should be a warning that the display reports itself as uncontrollable so if it is in fact not working, users do not open a ticket in vain. But we can get rid of it now, I'll expand on that later.

File: MonitorControl/AppDelegate.swift
Comment (waydabber) [reply MonitorControl/AppDelegate.swift:165]: Yes. We can increase the limit, I just didn't want to have some kind of freak thing when some app or dock makes tons of virtual displays and things go wrong.

File: MonitorControl/Manager/DisplayManager.swift
Comment (waydabber) [reply MonitorControl/Manager/DisplayManager.swift:2]: Absolutely nothing. I had something else in mind I think, or I donno what I wanted. I was distracted probably? :)

File: README.md
Comment (waydabber) [reply README.md:51]: Yes, I agree.

Comment (waydabber) [reply README.md:51]: I added this to the readme:

- macOS Mojave (`10.14`) and up _(note: you can download [version 2.1.0](https://github.com/MonitorControl/MonitorControl/releases/tag/v2.1.0) for macOS Sierra `10.12` support)_

File: MonitorControl/View Controllers/AboutPrefsViewController.swift
Comment (waydabber) [reply MonitorControl/View Controllers/AboutPrefsViewController.swift]: We can mark it but if we don't, it's probably ok. We'll know from the build number anyway.

File: MonitorControl/Manager/DisplayManager.swift
Comment (waydabber) [reply MonitorControl/Manager/DisplayManager.swift:2]: I removed it.

**Comment by waydabber** [comment]
Hi @JoniVR 

I removed the unused images and resolved the issues you mentioned. If you are good to go you can do the merge. I think it's simpler to proceed to an RC1 with the optimization hack and have the issue fixed without hurry for RC2 or the final release. But it's up to you!

Thank you!

**Comment by JoniVR** [comment]
image
2 Last things:
* Should we add a link to all collaborators on Github on the about tab? (perhaps below our names?), seems like a minor things but outside collaborators do help out quite a lot.
* The Contribute button, should we let it go to the actual Github, the one we have now or the one we have on the repo (just the base OpenCollective url)

@the0neyouseek @waydabber thoughts? 

I'll try to get this merged and get release candidate out later today.

**Comment by waydabber** [comment]
Yes, adding a link to contributors is a great idea. You mean this one? https://github.com/MonitorControl/MonitorControl/graphs/contributors

I am not entierly sure about the destination of the Contribute button. To be honest, I feel that the donate page is more straightforward than the general landing page where a less determined user might get lost in an instant, which is not exactly what we want. :) But it's up to you guys!

**Comment by JoniVR** [comment]
> Yes, adding a link to contributors is a great idea. You mean this one? MonitorControl/MonitorControl/graphs/contributors

Yes, or we could search for a tool that provides a nice overview, but I think that one for now will do.

> I am not entierly sure about the destination of the Contribute button. To be honest, I feel that the donate page is more straightforward than the general landing page where a less determined user might get lost in an instant, which is not exactly what we want. :) But it's up to you guys!

This is a bit of a special one imo, because to me, contributors can also be people who contribute code, so we could also rename it to "Donate" or something if we want it to refer to OpenCollective, or replace it with a link to the Repo, where donate buttons are visible and we kinda get both, but less directly.

**Comment by waydabber** [comment]
Ok, I'll add a link to the Contributors page and rename the button to Donate! I'll push it in an instant!

**Comment by waydabber** [comment]
It would look like this. But I need a bit more time sorry as I cannot solve the problem of the mouse not changing into a hand style over the link lol. :)

Screen Shot 2021-08-21 at 12 32 05

**Comment by waydabber** [comment]
Ok, whatever... It will not have a hand cursor. Tried like 4 techniques and I simply cannot change the cursor to look like a hand over that piece of text amazingly lol.

**Comment by waydabber** [comment]
@JoniVR - i committed the changes. You can go on! :)

**Review by JoniVR — APPROVED** [review]
All Good, fantastic work @waydabber! Will work on getting it released in the next 2 hours, have some other stuff going on. 

I'll also create a v3.1.0 branch once it's merged in, and then perhaps we should setup Github Projects with milestones so we can track and plan for v3.1.0?

**Comment by waydabber** [comment]
Great, thank you! :)

**Comment by waydabber** [comment]
Btw, I agree with the milestone approach, It would be great to track which issue is fixed in which release and also have some kind of general idea where the development is heading.

Whenever I have the time I am trying to go through the issues backlog, reply and close those that we can't do much about so we only have relevant issues and feature requests. Its such a heavy work though since we had so many issues open. Still, there is some progress, the number of open issues is now below 100 finally! :)

I think it would be great to somehow recruit some expert community members who try to help out others as a first line of defense so only those issues are escalated to devs which are really issues or valid requests.

**Comment by JoniVR** [comment]
> Btw, I agree with the milestone approach, It would be great to track which issue is fixed in which release and also have some kind of general idea where the development is heading.

Great, I'll get it set up!

> Whenever I have the time I am trying to go through the issues backlog, reply and close those that we can't do much about so we only have relevant issues and feature requests. Its such a heavy work though since we had so many issues open. Still, there is some progress, the number of open issues is now below 100 finally! :)

I think it would be great to somehow recruit some expert community members who try to help out others as a first line of defense so only those issues are escalated to devs which are really issues or valid requests.

Yeah, I tried for a long time to keep the number of issues in check, but at one point we were getting insane amounts of new users and issues and I just couldn't keep up anymore. I feel like we didn't grow contributors in correlation to users, which is why we got overwhelmed, I'm very glad to have you with us now and I agree we should probably soon look for some extra people to help.

**Comment by JoniVR** [comment]
As a side note, I think the amount of new issues got decreased quite a bit by the new issue templates, since they are much clearer to users and force them to actually read and fill out the templates!

**Pull request merged**
